File: x_b924(love_drawing)/C++/love_drawing.cpp
#include <iostream>

int main(int argc, char const *argv[]) {
  std::ios::sync_with_stdio(false);
  std::cin.tie(0);
  int vertex, side;
  while (std::cin >> vertex >> side) {
    int side_about_vertex[vertex + 1] = {0};
    int a, b;
    while (side--) {
      std::cin >> a >> b;
      side_about_vertex[a]++;
      side_about_vertex[b]++;
    }
    int odd = 0;
    for (int s : side_about_vertex) {
      odd += s % 2;
      if (odd > 2) break;
    }
    if (odd != 0 and odd != 2) std::cout << "NO" << '\n';
    else std::cout << "YES" << '\n';
  }
  return 0;
}

File: x_b924(love_drawing)/x_Python/love_drawing.py
from sys import stdin
while True:
    line = stdin.readline()
    if not line: break
    vertex, side = map(int, line.split())
    side_about_vertex = [0 for i in range(vertex + 1)]
    while side:
        a, b = map(int, stdin.readline().split())
        side_about_vertex[a] += 1
        side_about_vertex[b] += 1
        side -= 1
    odd = 0
    for side in side_about_vertex:
        odd += side % 2
        if odd > 2: break
    if odd != 0 and odd != 2: print('NO')
    else: print('YES')
